File: buff/classes/354/pub/ch8/excep.py
#!/usr/bin/python
# python excep.py

import sys

class ArgsException(Exception):
    def __init__(self,msg):
        self.msg=msg
    def __str__(self):
        return self.msg

def foo(i):
    if (i<2):
        raise ArgsException("too few args")
    # much more code
    print "foo() done"

try:
    foo(len(sys.argv))
    # much more code
    print "ok"
except ArgsException as e:
    print "not ok:",e

File: buff/classes/354/pub/ch8/Excep.java
// javac Excep.java && java Excep

class ArgsException extends Exception {
    private String msg;
    public ArgsException(String msg) { this.msg=msg; }
    public String toString()         { return msg; }
}

public class Excep {

    static void foo(int i) throws ArgsException {
	if (i<1)
	    throw new ArgsException("too few args");
	// much more code
	System.out.println("foo() done");
    }

    public static void main(String[] args) {
	try {
	    foo(args.length);
	    // much more code
	    System.out.println("ok");
	} catch (ArgsException e) {
	    System.out.println("not ok: "+e);
	}
    }

}
